templates/_common.tpl

```diff
@@ -99,11 +56,19 @@ tolerations:
 {{- end }}
 {{- end }}
 {{/*
-  coder.envproxyCredsEnv - temporary POC for intra-service auth
+  coder.envproxy.accessURL is a URL for accessing the envproxy.
 */}}
-{{- define "coder.envproxyCredsEnv" }}
-- name: "CODER_SERVER_USER"
-  value: {{ now | date "2006-01-02" | sha256sum }}
-- name: "CODER_SERVER_KEY"
-  value: {{ now | date "2006-01-02" | sha256sum }}
-{{- end }}
\ No newline at end of file
+{{- define "coder.envproxy.accessURL" }}
+{{- if ne .Values.envproxy.accessURL "" }}
+{{- .Values.envproxy.accessURL -}}
+{{- else if ne .Values.ingress.host "" }}
+    {{- if .Values.ingress.tls.enable -}}
+    https://
+    {{- else -}}
+    http://
+    {{- end -}}
+    {{- .Values.ingress.host }}/ingress
+{{- else }}
+{{- fail "envproxy.access.URL or ingress.host must be set" }}
+{{- end }}
+{{- end }}
```

Comment:
Does this implicitly not support raw IPs?

Reply:
If so, I think that's OK for alpha? But we will need to in the future, may have to poll kube for it on the ingress from envproxy itself or something on start.

Reply:
Yeah to support the raw IP of the load balancer you'd have to do it like that. Alternatively, the user could deploy with a fake envproxy access URL, wait for load balancer IP, and redeploy with real IP.